notifications_utils/clients/otel/otel_client.py

```diff
@@ -0,0 +1,179 @@
+        app.logger.info("OpenTelemetry metrics and spans will not be exported")
+        return
+    else:
+        raise ValueError(f"Invalid OTEL_EXPORT_TYPE: {export_mode}. Expected 'console', 'otlp', or 'none'.")
+
+    resource = Resource.create(
+        {"service.name": os.getenv("NOTIFY_APP_NAME") or app.config.get("NOTIFY_APP_NAME") or "notifications"}
+    )
+
+    provider = MeterProvider(metric_readers=metric_readers, resource=resource)
+    metrics.set_meter_provider(provider)
+
+    set_tracer_provider(TracerProvider(resource=resource))
+
+    def regex_predicate(baggage_key: str) -> bool:
+        return baggage_key.startswith("^public-.+")
+
+    get_tracer_provider().add_span_processor(BaggageSpanProcessor(regex_predicate))
+    get_tracer_provider().add_span_processor(span_processor)
+
+    _instrument_app(app)
+
+    app._otel_instrumented = True
+
+
+def _instrument_app(app: Flask) -> None:
+    """
+    Apply OpenTelemetry instrumentation based on available optional dependencies.
+    """
+
+    # Affects both requests and Flask instrumentation
+    os.environ["OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SERVER_REQUEST"] = app.config.get(
+        "OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SERVER_REQUEST", ".*"
+    )
+
+    os.environ["OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SERVER_RESPONSE"] = app.config.get(
+        "OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SERVER_RESPONSE", ".*"
+    )
+
+    os.environ["OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SANITIZE_FIELDS"] = app.config.get(
+        "OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SANITIZE_FIELDS", "authorization,set-cookie,.*session.*"
+    )
+
+    instrument_map = {
+        "celery": (_instrument_celery, "opentelemetry.instrumentation.celery"),
+        "flask": (_instrument_flask, "opentelemetry.instrumentation.flask"),
+        "redis": (_instrument_redis, "opentelemetry.instrumentation.redis"),
+        "requests": (_instrument_requests, "opentelemetry.instrumentation.requests"),
+        "sqlalchemy": (_instrument_sqlalchemy, "opentelemetry.instrumentation.sqlalchemy"),
+        "botocore": (_instrument_botocore, "opentelemetry.instrumentation.botocore"),
+    }
+
+    for name, (func, module_name) in instrument_map.items():
+        try:
+            __import__(module_name)
+            func(app)
+            app.logger.info("Enabled OpenTelemetry instrumentation: %s", name)
+        except ImportError:
+            app.logger.debug("Optional instrumentation '%s' not installed, skipping.", name)
+
+
+def _instrument_celery(app: Flask) -> None:
+    from opentelemetry.instrumentation.celery import CeleryInstrumentor
+
+    CeleryInstrumentor().instrument()
+
+
+def _instrument_flask(app: Flask) -> None:
+    from opentelemetry.instrumentation.flask import FlaskInstrumentor
+
+    FlaskInstrumentor().instrument_app(
+        app,
+        excluded_urls=app.config.get("OTEL_PYTHON_FLASK_EXCLUDED_URLS", "_status(\\?.*)?$,/metrics"),
```

Comment:
Is this env var only heeded by us? Might it be clearest if we prefix all the ones that are just consumed by our own code with `NOTIFY_`, even if it means we have a bunch of `NOTIFY_OTEL_..` env vars?

Reply:
No I think you've taken this too far. I just asked if we wanted _environment variable_ names that _we've invented_ to be prefixed with `NOTIFY_`. Certainly don't want e.g. things named `NOTIFY_OTEL_METRIC_EXPORT_INTERVAL` when otel natively consumes `OTEL_METRIC_EXPORT_INTERVAL`

Reply:
It doesn't make any sense. Maybe pretend I didn't say anything.

Reply:
Okay I've removed the 'NOTIFY_'.